src/PrintBox_text.ml

```diff
@@ -345,6 +348,15 @@ module Box_inner = struct
     | Pad (dim, b') ->
       let expected_size = size b in
       render_rec ~offset:Pos.(dim + offset) ~expected_size ~out b' Pos.(pos + dim)
+    | Align_right b' ->
+      begin match expected_size with
+        | Some expected_size ->
+          let left_pad = expected_size.x - (size b').x in
+          let pos' = Pos.move_x pos left_pad in
+          render_rec ~offset ~expected_size ~out b' pos'
```

Comment:
Do I need to modify `offset` or `expected_size` in the recursive call here?

Reply:
I don't think `expected_size` should change, but for `offset` I don't remember. I recommend writing tons of tests (mdx!) and seeing how it behaves :p